src/main/java/org/magicdgs/readtools/tools/distmap/DistmapPartDownloader.java

```diff
@@ -0,0 +1,235 @@
+            doc = "Validation stringency for all SAM/BAM/CRAM files read by this program. "
+                    + "The default stringency value SILENT can improve performance when processing "
+                    + "a BAM file in which variable-length data (read, qualities, tags) do not otherwise need to be decoded.",
+            common = true, optional = true)
+    private ValidationStringency readValidationStringency =
+            ReadConstants.DEFAULT_READ_VALIDATION_STRINGENCY;
+
+    // SamReaderFactory constructed on demand with the parameters
+    private SamReaderFactory factory = null;
+
+    private SamReaderFactory getSamReaderFactory() {
+        if (factory == null) {
+            factory = SamReaderFactory.makeDefault()
+                    .referenceSequence(referenceFile)
+                    .validationStringency(readValidationStringency);
+        }
+        return factory;
+    }
+
+    public void downloadParts(final List<Path> partFiles, final Function<SAMFileHeader, SAMProgramRecord> programRecord) {
+        Utils.nonEmpty(partFiles);
+        outputArgumentCollection.validateUserOutput();
+
+        // for the final merging
+        final ReadsDataSource toMerge;
+        final boolean presorted;
+
+        if(partFiles.size() <= numberOfBatches) {
+            // logging the process and start the progress meter
+            logger.info(
+                    "Only {} parts found: download will be performed at the same time as merging",
+                    partFiles::size);
+            toMerge = new ReadsDataSource(partFiles, getSamReaderFactory());
+            presorted = isPresorted();
+        } else {
+            toMerge = downloadBatchesAndPreSort(partFiles);
+            presorted = true;
+        }
+
+        final SAMFileHeader header = setHeaderOptions(toMerge.getHeader());
+
+        // initialize the progress meter
+        logger.info("Merging output");
+        final ProgressMeter mergingProgress = new ProgressMeter(secondsBetweenProgressUpdates);
+        mergingProgress.start();
+
+        writeReads(toMerge,
+                outputArgumentCollection.outputWriter(header, () -> programRecord.apply(header),
+                        presorted, referenceFile),
+                mergingProgress);
+
+        // logging the end
+        mergingProgress.stop();
+        logger.debug("Finished download and merging.", partFiles.size());
+    }
+
+    private boolean isPresorted() {
+        // TODO: this should also include the header to check if they are sorted in the same order
+        // TODO: nevertheless, GATK ReadsDataSource returns a merged header with coordinate sort order
+        // TODO: and it is impossible to distinguish when it is pre-sorted
+        return SAMFileHeader.SortOrder.unsorted == sortOrder;
+    }
+
+    private SAMFileHeader setHeaderOptions(final SAMFileHeader header) {
+        header.setSortOrder(sortOrder);
+        if (!noDropProgramGroups) {
```

Comment:
Add an else clause with a warning showing that the program groups are going to be output, and that this is going to output lot of of `@PG` header lines...